### PR TITLE
fix: sort consortia on collection create or edit (#3996)

### PR DESCRIPTION
**Pull request by NoopDog**
- #TICKET_NUMBER
#3996 

### Reviewers
**Functional:** 
@ebezzi 

---

## Changes
- sorted collection during creation and edit of consortia
- added tests for portal and curation APIs

### CODE REVIEW

File: backend/layers/common/entities.py [hunk 1]
```diff
@@ -223,6 +223,14 @@ def strip_fields(self):
         if self.consortia is not None:
             self.consortia = [consortium.strip() for consortium in self.consortia]
 
+    def sortConsortia(self):
+        if self.consortia is not None:
+            self.consortia = sorted(self.consortia)
```
Comment (ebezzi) [inline backend/layers/common/entities.py:228]: Nit: you can do `self.consortia.sort()` and will sort in place

File: backend/layers/common/entities.py [hunk 2]
```diff
@@ -223,6 +223,14 @@ def strip_fields(self):
         if self.consortia is not None:
             self.consortia = [consortium.strip() for consortium in self.consortia]
 
+    def sortConsortia(self):
```
Comment (ebezzi) [inline backend/layers/common/entities.py:226]: Use snake case here: `sort_consortia`

**Review by ebezzi — CHANGES_REQUESTED** [review]

File: backend/layers/common/entities.py (hunk 2 above)
Comment (NoopDog) [reply backend/layers/common/entities.py]: Updated, thanks.

File: backend/layers/common/entities.py (hunk 1 above)
Comment (NoopDog) [reply backend/layers/common/entities.py]: Updated!

**Review by ebezzi — APPROVED** [review]

**Pull request merged**